cmd/check_changes.cpp

```diff
@@ -64,88 +64,95 @@ int main(int argc, char* argv[]) {
     CLI11_PARSE(app, argc, argv);
 
     absl::Time t1{absl::Now()};
-    std::cout << t1 << " Checking change sets in " << chaindata << "\n";
-
-    db::EnvConfig db_config{chaindata};
-    auto env{db::open_env(db_config)};
-    auto txn{env.start_read()};
-    auto chain_config{db::read_chain_config(txn)};
-    if (!chain_config) {
-        throw std::runtime_error("Unable to retrieve chain config");
-    }
 
-    AnalysisCache analysis_cache;
-    ExecutionStatePool state_pool;
+    SILKWORM_LOG(LogLevel::Info) << " Checking change sets in " << chaindata << "\n";
 
     uint64_t block_num{from};
-    for (; block_num < to; ++block_num) {
 
-        txn.renew_reading();
-        std::optional<BlockWithHash> bh{db::read_block(txn, block_num, /*read_senders=*/true)};
-        if (!bh) {
-            break;
+    try {
+        db::EnvConfig db_config{chaindata};
+        auto env{db::open_env(db_config)};
+        auto txn{env.start_read()};
+        auto chain_config{db::read_chain_config(txn)};
+        if (!chain_config) {
+            throw std::runtime_error("Unable to retrieve chain config");
         }
 
-        db::Buffer buffer{txn, block_num};
+        AnalysisCache analysis_cache;
+        ExecutionStatePool state_pool;
 
-        ValidationResult err{execute_block(bh->block, buffer, *chain_config, &analysis_cache, &state_pool).second};
-        if (err != ValidationResult::kOk) {
-            std::cerr << "Failed to execute block " << block_num << "\n";
-            continue;
-        }
+        for (; block_num < to; ++block_num) {
+            txn.renew_reading();
+            std::optional<BlockWithHash> bh{db::read_block(txn, block_num, /*read_senders=*/true)};
+            if (!bh) {
+                break;
+            }
+
+            db::Buffer buffer{txn, block_num};
 
-        db::AccountChanges db_account_changes{db::read_account_changes(txn, block_num)};
-        const db::AccountChanges& calculated_account_changes{buffer.account_changes().at(block_num)};
-        if (calculated_account_changes != db_account_changes) {
-            bool mismatch{false};
+            ValidationResult err{execute_block(bh->block, buffer, *chain_config, &analysis_cache, &state_pool).second};
+            if (err != ValidationResult::kOk) {
+                SILKWORM_LOG(LogLevel::Error) << "Failed to execute block " << block_num << std::endl;
```

Comment:
There's no difference between `"\n"` and `std::endl` since `std::cerr` is flushed immediately.